rskj-core/src/test/java/co/rsk/RskContextTest.java

```diff
@@ -43,25 +60,150 @@ public void getCliArgsSmokeTest() {
+        doReturn(new GarbageCollectorConfig(true, 1000, (int) numberOfEpochs)).when(testProperties).garbageCollectorConfig();
+        doReturn(testDatabasesDirectory.toString()).when(testProperties).databaseDir();
+
+        RskContext rskContext = new RskContext(new String[0]) {
+            @Override
+            public RskSystemProperties getRskSystemProperties() {
+                return testProperties;
+            }
+        };
+
+        Files.createDirectory(testDatabasesDirectory.resolve("unitrie"));
+
+        TrieStore trieStore = rskContext.getTrieStore();
+        Assert.assertThat(trieStore, is(instanceOf(MultiTrieStore.class)));
+        Assert.assertThat(Files.list(testDatabasesDirectory).count(), is(numberOfEpochs));
+        Assert.assertThat(Files.list(testDatabasesDirectory).noneMatch(p -> p.getFileName().toString().equals("unitrie")), is(true));
+    }
+
+    @Test(expected = IllegalStateException.class)
+    @Ignore("Permissions set fails under CircleCI")
+    public void shouldFailIfCannotBuildMultiTrieStore() throws IOException {
+        Path testDatabasesDirectory = Files.createTempDirectory("test", PosixFilePermissions.asFileAttribute(PosixFilePermissions.fromString("---------")));
+        doReturn(new GarbageCollectorConfig(true, 1000, 3)).when(testProperties).garbageCollectorConfig();
+        doReturn(testDatabasesDirectory.toString()).when(testProperties).databaseDir();
+
+        RskContext rskContext = new RskContext(new String[0]) {
+            @Override
+            public RskSystemProperties getRskSystemProperties() {
+                return testProperties;
+            }
+        };
+
+        rskContext.getTrieStore();
     }
 
-    private void mockBuildInfoResource(InputStream buildInfoStream) {
-        mockStatic(RskContext.class);
-        ClassLoader classLoader = mock(ClassLoader.class);
-        when(classLoader.getResourceAsStream("build-info.properties")).thenReturn(buildInfoStream);
-        when(RskContext.class.getClassLoader()).thenReturn(classLoader);
+    @Test
+    public void mergeMultiTrieStoreDBs() throws IOException {
+        Path testDatabasesDirectory = databaseDir.getRoot().toPath();
+
+        List<Path> sourcePaths = new ArrayList<>();
+        Set<byte[]> sourceKeys = new HashSet<>();
+        int sourcesCount = 3;
+        for (int i = 0; i < sourcesCount; i++) {
+            Path sourcePath = testDatabasesDirectory.resolve(String.format("src-%d", i));
+            sourcePaths.add(sourcePath);
+            KeyValueDataSource originDataSource = RskContext.makeDataSource(sourcePath);
+            byte[] currentElement = {(byte) i};
+            sourceKeys.add(currentElement);
+            originDataSource.put(currentElement, currentElement);
+            originDataSource.close();
+        }
+
+        Path destination = testDatabasesDirectory.resolve("destination");
+        RskContext.mergeMultiTrieStoreDBs(destination, sourcePaths);
```

Comment:
Please move this method out of RskContext to MultiTrieStore or TrieStoreImpl

Reply:
Given that `MultiTrieStore` or `TrieStoreImpl` doesn't have visibility of any kind of `java.util.Path`, I moved it to `LevelDbDataSource#mergeDataSources`